src/markdown-pages/docs/ecosystem/osgi-bundle-repository.md

```diff
@@ -0,0 +1,156 @@
+---
+path: "/docs/ecosystem/osgi-bundle-repository"
+title: "OSGi Bundle Repository (OBR)"
+---
+
+# OSGi Bundle Repository (OBR) in Galasa
+
+OSGi Bundle Repository (OBR) is a service that simplifies deploying and using OSGi bundles in Galasa. It provides automatic dependency resolution and deployment of bundles from repositories.
+
+## Key Benefits
+
+- Automatic installation of bundles with their dependencies
+- Access to both executable bundles and source code
+- Simplified bundle deployment and updates
+- Support for local and remote bundle repositories
+
+## Key Components
+
+- **Repository Admin** - Service to access bundle repositories
+- **Repository** - Provides access to available bundles
+- **Resource** - Description of a bundle to be installed
+- **Resolver** - Handles bundle dependency resolution and deployment
+
+## Using OBR in Galasa
+
+In Galasa, OBR is primarily used behind the scenes to manage test bundles and their dependencies. The framework handles most OBR interactions automatically, so you typically don't need to work with it directly.
+
+When you build Galasa tests, they are packaged as OSGi bundles and can be stored in an OBR repository. This allows Galasa to properly resolve and load all required dependencies when running tests.
+
+## How OBR Works in Galasa
+
+### Repository Structure
+OBR uses XML-based repository files to describe available bundles and their metadata. These files contain:
+- Bundle descriptions and documentation
+- Package dependencies and version requirements  
+- Service capabilities and requirements
+- Resource capabilities and constraints
+
+### Bundle Resolution
+When Galasa needs to resolve bundle dependencies:
+
+1. The OBR service searches available repositories
+2. Analyzes bundle requirements and capabilities 
+3. Constructs a dependency graph
+4. Determines the optimal set of bundles to satisfy all dependencies
+5. Downloads and installs required bundles
+
+### Common Use Cases
+
+#### Local Development
+When developing Galasa tests locally:
+- Test bundles are built and stored in your local Maven repository
+- OBR indexes these bundles automatically
+- Dependencies are resolved from both local and remote repositories
+- Changes to test bundles are immediately available
+
+#### CI/CD Environments  
+In automated environments:
+- Test bundles are published to shared repositories
+- OBR handles coordinated deployment across test runs
+- Version conflicts are automatically resolved
+- Bundle updates are managed consistently
+
+## Repository Management
+
+### Adding Local Repositories
+You can create local repositories for your custom bundles:
+
+```properties
+# Example repository.xml
+<repository name="MyTestRepo" ...>
+    <resource>
+        <capability name='bundle'>
```

Comment:
I'm not certain what this example is showing or how the `n` and `t` attributes fit in. Perhaps more comments or a description before the xml snippet would help?